File: c#/time-deposit-kata-net/TimeDepositCalculator.cs
using System;
using System.Collections.Generic;

namespace time_deposit_kata_net
{
    public class TimeDepositCalculator
    {
        public void UpdateBalance(List<TimeDeposit> temps)
        {
            for (int i = 0; i < temps.Count; i++)
            {
                double interest = 0;

                if (temps[i].Days > 30)
                {
                    if (temps[i].PlanType == "student")
                    {
                        if (temps[i].Days < 366)
                        {
                            interest += temps[i].Balance * 0.03 / 12;
                        }
                    }
                    else if (temps[i].PlanType == "premium")
                    {
                        if (temps[i].Days > 45)
                        {
                            interest += temps[i].Balance * 0.05 / 12;
                        }
                    }
                    else if (temps[i].PlanType == "basic")
                    {
                        interest += temps[i].Balance * 0.01 / 12;
                    }
                }

                temps[i].Balance += Math.Round(interest, 2);
            }
        }
    }
}

File: java/src/main/java/org/ikigaidigital/service/TimeDepositCalculator.java
package org.ikigaidigital;

import java.math.BigDecimal;
import java.math.RoundingMode;
import java.util.List;

public class TimeDepositCalculator {
    public void updateBalance(List<TimeDeposit> xs) {
        for (int i = 0; i < xs.size(); i++) {
            double interest = 0;

            if (xs.get(i).getDays() > 30) {
                if (xs.get(i).getPlanType().equals("student")) {
                    if (xs.get(i).getDays() < 366) {
                        interest += xs.get(i).getBalance() * 0.03 / 12;
                    }
                } else if (xs.get(i).getPlanType().equals("premium")) {
                    if (xs.get(i).getDays() > 45) {
                        interest += xs.get(i).getBalance() * 0.05 / 12;
                    }
                } else if (xs.get(i).getPlanType().equals("basic")) {
                    interest += xs.get(i).getBalance() * 0.01 / 12;
                }
            }

            double a2d = xs.get(i).getBalance() + (new BigDecimal(interest).setScale(2, RoundingMode.HALF_UP)).doubleValue();
            xs.get(i).setBalance(a2d);
        }
    }
}
